js/test/Ice/defaultServant/Client.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) ZeroC, Inc.
+
+import { Ice } from "@zeroc/ice";
+import { TestHelper } from "../../Common/TestHelper.js";
+import { Test } from "./Test.js";
+const test = TestHelper.test;
+
+export class Client extends TestHelper {
+    async allTests(args: string[]) {
+        const communicator = this.communicator();
+        const oa = new Test.RemoteObjectAdapterPrx(communicator, `remote-object-adapter:${this.getTestEndpoint()}`);
```

Comment:
I don't understand why we need this complicated setup. We can't simply host an OA locally and call it directly?

Reply:
With JavaScript we use the C++ Echo Server from `cpp/test/Ice/echo` to workaround the fact that the JavaScript mapping doesn't support incoming connections. The test scripts takes care of automatically starting the EchoServer for JavaScript ClientServerTests.

If we want to use a client test only, we can add a custom test.py script that ensures echo server is also started.

Reply:
I meant: we can keep this client -> c++ server -> back to same JS client.

But when the client need to call its own local OA, it can just call it directly. It does not need to make remote invocations via the C++ server.

Reply:
Simplified as suggested.